File: src/test/java/org/zenbot/szolnok/timetable/batch/step/bus/reader/UrlResourceItemReaderTest.java
package org.zenbot.szolnok.timetable.batch.step.bus.reader;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.zenbot.szolnok.timetable.batch.step.readurls.StringResourcesInMemoryStorage;

import static org.mockito.Mockito.times;
import static org.mockito.Mockito.verify;
import static org.mockito.MockitoAnnotations.initMocks;

public class UrlResourceItemReaderTest {

    @Mock
    private StringResourcesInMemoryStorage stringResourcesInMemoryStorage;

    private UrlResourceItemReader testSubject;

    @Before
    public void setUp() {
        initMocks(this);
        testSubject = new UrlResourceItemReader(stringResourcesInMemoryStorage);
    }

    @Test
    public void readShouldCallTheStringResourcesInMemoryStorageEachTime() {
        // GIVEN


        // WHEN
        testSubject.read();
        testSubject.read();

        //THEN
        verify(stringResourcesInMemoryStorage, times(2)).get();
    }

}

File: szolnok-timetable-backend-batch-utils/src/test/kotlin/org/zenbot/szolnok/timetable/backend/batch/utils/common/batch/reader/UrlResourceItemReaderTest.kt
package org.zenbot.szolnok.timetable.backend.batch.utils.common.batch.reader

import org.assertj.core.api.Assertions.assertThat
import org.junit.Before
import org.junit.Test
import org.mockito.BDDMockito.given
import org.mockito.BDDMockito.verify
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.MockitoAnnotations
import org.zenbot.szolnok.timetable.backend.batch.utils.common.service.StringResourcesInMemoryStorage

class UrlResourceItemReaderTest {

    @InjectMocks
    private lateinit var testSubject: UrlResourceItemReader

    @Mock
    private lateinit var stringResourcesInMemoryStorage: StringResourcesInMemoryStorage

    @Before
    fun setUp() {
        MockitoAnnotations.initMocks(this)
    }

    @Test
    fun `read should call the stringResourcesInMemoryStorage`() {
        // GIVEN
        val getResult = "result"
        given(stringResourcesInMemoryStorage.get()).willReturn(getResult)

        // WHEN
        val result = testSubject.read()

        // THEN
        verify(stringResourcesInMemoryStorage).get()

        assertThat(result)
                .isNotNull()
                .isSameAs(getResult)
    }
}
